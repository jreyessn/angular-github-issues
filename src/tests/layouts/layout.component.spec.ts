import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatToolbarModule } from '@angular/material/toolbar';
import { RouterTestingModule } from '@angular/router/testing';

import { LayoutComponent } from '../../app/layouts/layout.component';

describe('LayoutComponent', () => {
  let component: LayoutComponent;
  let fixture: ComponentFixture<LayoutComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ LayoutComponent ],
      imports: [
        MatToolbarModule,
        MatIconModule,
        MatButtonModule,
        RouterTestingModule
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LayoutComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('Debe hacer el match con el snapshot', () => {
    expect(compiled).toMatchSnapshot()
  });

});
