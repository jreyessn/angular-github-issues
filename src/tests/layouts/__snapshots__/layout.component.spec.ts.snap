// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent Debe hacer el match con el snapshot 1`] = `
<div
  id="root1"
>
  <mat-toolbar
    class="mat-toolbar mat-primary mat-toolbar-single-row"
    color="primary"
  >
    <button
      aria-label="icon-button with menu icon"
      class="mat-focus-indicator mat-icon-button mat-button-base"
      mat-icon-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          search
        </mat-icon>
      </span>
      <span
        class="mat-ripple mat-button-ripple mat-button-ripple-round"
        matripple=""
      />
      <span
        class="mat-button-focus-overlay"
      />
    </button>
    <span
      class="mx-2"
    >
      Github Issues Search
    </span>
  </mat-toolbar>
  <div
    class="flex flex-col flex-auto p-6 sm:p-14"
  >
    <router-outlet />
    
  </div>
</div>
`;
